src/Admin/SiteHealth.php

```diff
@@ -132,6 +171,37 @@ public function add_tests( $tests ) {
 			'test'  => [ $this, 'xdebug_extension' ],
 		];
 
+		// Only add the async test if on WP 5.6+ and *not* on version of Health Check plugin which doesn't support REST async tests.
+		$supports_async = version_compare( get_bloginfo( 'version' ), '5.6', '>=' );
```

Comment:
@westonruter Async tests have always been apart of Site Health, just that they've been done via AJAX. It was [since WP 5.6](https://github.com/WordPress/wordpress-develop/blob/092ee0fbffaa8723cbddd1134ec811fdcf8601b8/src/wp-admin/includes/class-wp-site-health.php#L2262) async tests could be done via a REST endpoint, so I don't think we need to restrict the test to only WP > 5.6.

Reply:
Yeah, but the `has_rest` is only recognized as of WP 5.6, so that's why I'm limiting it to that.

Reply:
Making it async via AJAX would provide the most compatibility then, no? AFAIK the Health Check plugin does not support `has_rest` either.

Reply:
Yeah, but the REST implementation is much better and we don't fully support WP prior to 5.6 anyway. The Health Check plugin doesn't support `has_rest` but only because it is out of date. The released version on WordPress.org doesn't support it, but the unreleased version on GitHub _does_ as I found in https://github.com/ampproject/amp-wp/pull/6456#discussion_r706502388.

Reply:
Ah I see, that makes sense.